modules/ximgproc/samples/thinning.cpp

```diff
@@ -10,7 +10,7 @@ using namespace cv;
 
 int main()
 {
-    Mat img = imread("opencv-logo.png", IMREAD_COLOR);
+    Mat img = imread("../data/opencv-logo.png", IMREAD_REDUCED_COLOR_2);
```

Comment:
Well it is a matter of flavor. 
About the reduced flag, I would not do this, it is additional functionality that is NOT needed in this case and could confuse people.